overlord/ifacestate/handlers.go

```diff
@@ -458,13 +444,9 @@ func (m *InterfaceManager) doDisconnect(task *state.Task, _ *tomb.Tomb) error {
 		return err
 	}
 
-	affectedConns, err := m.repo.ResolveDisconnect(plugRef.Snap, plugRef.Name, slotRef.Snap, slotRef.Name)
```

Comment:
One thing that we might see (this is probably very unlikely) is new snapd handling a task that was created by old snapd. If that were to happen this code will misbehave.